--- strongswan-5.9.14/src/libstrongswan/plugins/openssl/openssl_kdf.c.orig	2026-01-08 08:42:02.236725459 -0500
+++ strongswan-5.9.14/src/libstrongswan/plugins/openssl/openssl_kdf.c	2026-01-08 08:44:19.315520163 -0500
@@ -201,6 +201,14 @@
 		.key = chunk_clone(chunk_from_str("00000000000000000000000000000000")),
 	);
 
+	/* also generate a salt (as if none was provided, i.e. zeroes of hash length)
+	* as OpenSSL 3.5.1+ won't accept NULL anymore */
+	if (algo == KDF_PRF)
+	{
+		this->salt = chunk_copy_pad(chunk_alloc(get_length(this)),
+								chunk_empty, 0);
+	}
+
 	if (!this->hasher ||
 		!get_bytes(this, algo == KDF_PRF ? get_length(this) : sizeof(buf), buf))
 	{
