--- python-pysnmp-5.0.20/pysnmp/smi/builder.py.orig	2025-02-10 16:21:22.582635683 -0500
+++ python-pysnmp-5.0.20/pysnmp/smi/builder.py	2025-02-10 16:23:01.083627562 -0500
@@ -10,26 +10,15 @@
 import marshal
 import time
 import traceback
+import importlib.util
+import importlib.machinery
 
 try:
-    import importlib
-
-    try:
-        PY_MAGIC_NUMBER = importlib.util.MAGIC_NUMBER
-        SOURCE_SUFFIXES = importlib.machinery.SOURCE_SUFFIXES
-        BYTECODE_SUFFIXES = importlib.machinery.BYTECODE_SUFFIXES
-
-    except Exception:
-        raise ImportError()
-
-except ImportError:
-    import imp
-
-    PY_MAGIC_NUMBER = imp.get_magic()
-    SOURCE_SUFFIXES = [s[0] for s in imp.get_suffixes()
-                       if s[2] == imp.PY_SOURCE]
-    BYTECODE_SUFFIXES = [s[0] for s in imp.get_suffixes()
-                         if s[2] == imp.PY_COMPILED]
+    PY_MAGIC_NUMBER = importlib.util.MAGIC_NUMBER
+    SOURCE_SUFFIXES = importlib.machinery.SOURCE_SUFFIXES
+    BYTECODE_SUFFIXES = importlib.machinery.BYTECODE_SUFFIXES
+except Exception:
+    raise ImportError()
 
 PY_SUFFIXES = SOURCE_SUFFIXES + BYTECODE_SUFFIXES
 
