--- flex-2.6.4/lib/malloc.c.orig	2025-05-06 15:20:58.199783950 -0400
+++ flex-2.6.4/lib/malloc.c	2025-05-06 15:21:10.615082525 -0400
@@ -1,13 +1,13 @@
      #include <config.h>
      #undef malloc
-     
+
      #include <sys/types.h>
-     
-     void *malloc ();
-     
+
+     void *malloc (size_t n);
+
      /* Allocate an N-byte block of memory from the heap.
         If N is zero, allocate a 1-byte block.  */
-     
+
      void *
      rpl_malloc (size_t n)
      {
