--- util-linux-2.41.1/login-utils/su-common.c.orig	2025-08-28 16:15:29.463656804 -0400
+++ util-linux-2.41.1/login-utils/su-common.c	2025-08-28 16:16:32.594296270 -0400
@@ -282,22 +282,6 @@
 }
 #endif

-/* Log the fact that someone has run su to the user given by PW;
-   if SUCCESSFUL is true, they gave the correct password, etc.  */
-
-static void log_syslog(struct su_context *su, bool successful)
-{
-	DBG(LOG, ul_debug("syslog logging"));
-
-	openlog(su->runuser ? "runuser" : "su", LOG_PID, LOG_AUTH);
-	syslog(LOG_NOTICE, "%s(to %s) %s on %s",
-	       successful ? "" :
-	       su->runuser ? "FAILED RUNUSER " : "FAILED SU ",
-	       su->new_user, su->old_user ? : "",
-	       su->tty_name ? : "none");
-	closelog();
-}
-
 /*
  * Log failed login attempts in _PATH_BTMP if that exists.
  */
@@ -436,8 +420,6 @@
 	}

  done:
-	log_syslog(su, !is_pam_failure(rc));
-
 	if (is_pam_failure(rc)) {
 		const char *msg;
