--- psplash-53ae74a36bf17675228552abb927d2f981940a6a/psplash-colors.h.orig	2026-04-02 15:34:39.126987715 -0400
+++ psplash-53ae74a36bf17675228552abb927d2f981940a6a/psplash-colors.h	2026-04-02 15:35:36.572504102 -0400
@@ -12,10 +12,10 @@
 #define _HAVE_PSPLASH_COLORS_H
 
 /* This is the overall background color */
-#define PSPLASH_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BACKGROUND_COLOR 0x00,0x00,0x00
 
 /* This is the color of any text output */
-#define PSPLASH_TEXT_COLOR 0x6d,0x6d,0x70
+#define PSPLASH_TEXT_COLOR 0xff,0xff,0xff
 
 /* This is the color of the progress bar indicator */
 #define PSPLASH_BAR_COLOR 0x6d,0x6d,0x70
