--- hostapd-2.11/src/ap/hostapd.c.orig	2024-11-07 10:25:00.624347754 -0500
+++ hostapd-2.11/src/ap/hostapd.c	2024-11-07 10:25:44.561423858 -0500
@@ -1552,7 +1552,7 @@
 						conf->ssid.ssid_len);
 
 	if (!hostapd_drv_none(hapd)) {
-		wpa_printf(MSG_DEBUG, "Using interface %s with hwaddr " MACSTR
+		wpa_printf(MSG_ERROR, "Using interface %s with hwaddr " MACSTR
 			   " and ssid \"%s\"",
 			   conf->iface, MAC2STR(hapd->own_addr),
 			   wpa_ssid_txt(conf->ssid.ssid, conf->ssid.ssid_len));
